### ISSUE TITLE
Reconsider criteria grouping for WCAG 3.0

### ISSUE THREAD

**Issue by Dan503** [w3c/wcag#1175]
When I was first trying to do accessibility audits on client projects, it was extremely difficult to use each guideline as a checkbox to say "does the website pass this or not" largely because of the grouping that has been used to group the criteria.

So for example, lets say that I am doing the keyboard testing for a website that I am doing an audit on. There are quite a few criteria related to keyboard navigability but they aren't all grouped together nicely.

So these are the criteria for 2 Operable. I'll highlight all the ones that are relevant when testing the keyboard accessibility:

- 2.1 Keyboard Accessible
    - **2.1.1 Keyboard (yes)**
    - **2.1.2 No Keyboard Trap (yes)**
    - **2.1.3 Keyboard (No Exception) (yes)**
    - **2.1.4 Character Key Shortcuts (yes)**
- 2.2 Enough Time
    - 2.2.1 Timing Adjustable
    - 2.2.2 Pause, Stop, Hide
    - 2.2.3 No Timing
    - 2.2.4 Interruptions
    - 2.2.5 Re-authenticating
    - 2.2.6 Timeouts
- 2.3 Seizures and Physical Reactions
    - 2.3.1 Three Flashes or Below Threshold
    - 2.3.2 Three Flashes
    - 2.3.3 Animation from Interactions
- 2.4 Navigable
    - **2.4.1 Bypass Blocks (yes)**
    - 2.4.2 Page Titled
    - **2.4.3 Focus Order (yes)**
    - 2.4.4 Link Purpose (In Context)
    - 2.4.5 Multiple Ways
    - 2.4.6 Headings and Labels
    - **2.4.7 Focus Visible (yes)**
    - 2.4.8 Location
    - 2.4.9 Link Purpose (Link Only)
    - 2.4.10 Section Headings
- 2.5 Input Modalities
    - 2.5.1 Pointer Gestures
    - 2.5.2 Pointer Cancellation
    - 2.5.3 Label in Name
    - 2.5.4 Motion Actuation
    - 2.5.5 Target Size
    - 2.5.6 Concurrent Input Mechanisms

Notice how there are keyboard testing related items under "2.4 Navigable" but there are 2 whole sections between 2.1 Keyboard Accessible and 2.4 Navigable that are completely unrelated to keyboard testing.

That means that when I do accessibility audits using the official WCAG order and grouping I have to scroll through pages and pages of notes to jump from 2.1 to 2.4 because those two sections were highly related to one another.

When doing later tests, I used a different method for grouping the guidelines that was based more on how I actually go about testing a website rather than the official order used by WCAG.

<details>
<summary>My custom WCAG grouping</summary>

I am by no means saying this is the definitive order. This is just a grouping that I use to help me do audits more easily.

- Multi-media content

    - 1.4.2 Audio Control
    - 1.2.1 Audio-only and Video-only (Prerecorded)
    - 1.2.2 Captions (Prerecorded)
    - 1.2.3 Audio Description or Media Alternative (Prerecorded)
    - 2.3.1 Three Flashes or Below Threshold - (video content is the most common reason this will fail)
    - 1.2.4 Captions (Live)
    - 1.2.5 Audio Description (Pre-recorded)
    - 1.2.6 Sign Language (Pre-recorded)
    - 1.2.7 Extended Audio description (Pre-recorded)
    - 1.2.8 Media Alternative (Pre-recorded)
    - 1.2.9 Audio-Only (Live)
    - 1.4.7 Low or No Background Audio
    - 2.3.2 Three Flashes or Below (strict)

- Keyboard navigability

    - 2.1.1 Keyboard
    - 2.1.2 No Keyboard Trap
    - 2.4.1 Bypass Blocks
    - 2.4.3 Focus Order
    - 2.1.4 Character key shortcuts
    - 2.4.7 Focus Visible
    - 2.1.3 Keyboard (No Exception)

- Predictability

    - 2.4.4 Link Purpose (In Context)
    - 3.2.1 On Focus
    - 3.2.3 Consistent Navigation
    - 3.2.4 Consistent Identification
    - 3.2.5 Change on Request
    - 2.2.4 Interruptions
    - 2.4.9 Link Purpose (Link Only)

- Screen reader friendliness

    - 1.1.1 Non-text Content
    - 1.3.1 "Info and relationships" (this one is really big and vague, I'd prefer to have it broken down more)
    - 1.3.2 Meaningful Sequence
    - 4.1.1 Parsing
    - 4.1.2 Name, Role, Value
    - 2.4.6 Headings and Labels
    - 4.1.3 Status Messages

- User friendliness

    - 2.4.2 Page Titled
    - 2.5.2 Pointer Cancellation
    - 2.4.5 Multiple Ways
    - 2.4.8 Location
    - 2.2.5 Re-authenticating
    - 2.5.6 Concurrent Input Mechanisms
    - 1.4.13 Content on Hover or Focus
    - Languages
        - 3.1.1 Language of Page
        - 3.1.2 Language of Parts
    - Control over timing
        - 2.2.1 Timing Adjustable
        - 2.2.2 Pause, Stop, Hide
        - 2.2.3 No Timing
        - 2.2.6 Timeouts
        - 2.3.3 Animation from Interactions
    - Understandable content
        - 2.4.10 Section Headings
        - 3.1.3 Unusual words
        - 3.1.4 Abbreviations
        - 3.1.5 Reading Level
        - 3.1.6 Pronunciation
    - 1.3.6 Identify Purpose

- Forms

    - 3.2.2 On Input
    - 2.5.3 Label in Name
    - 3.3.5 Help
    - 3.3.2 Labels or Instructions
    - Error Handling
        - 3.3.1 Error Identification
        - 3.3.3 Error Suggestion
        - 3.3.4 Error Prevention (Legal, Financial, Data)
        - 3.3.6 Error Prevention (All)
    - 1.3.5 Identify Input Purpose
        - Correct input types
        - Valid “autocomplete” attribute

- Visual and other sensory recognition

    - 1.3.3 Sensory Characteristics
    - 1.4.5 Images of Text
    - 1.4.9 Images of Text (No Exception)
    - 1.4.1 Use of Color
    - 1.4.3 Contrast (Minimum)
    - 1.4.11 Non-text Contrast
    - 1.4.6 Contrast (Enhanced)
    - User controlled font settings
        - 1.4.4 Resize text
        - 1.4.12 Text Spacing
    - 1.4.8 Visual Presentation
        - Customisable foreground and background colours
        - Paragraph width is never more than 80 characters wide
        - Text is not justified
        - Sufficient line and paragraph spacing
        - (Line-height minimum of [1.5 x font-size], Paragraph spacing is set to a minimum of [1.5 x line-height]).
        - Controls provided for changing font size

- Mobile friendliness

    - 1.3.4 Orientation
    - 1.4.10 Reflow
    - 2.5.1 Pointer Gestures
    - 2.5.4 Motion Actuation
    - 2.5.5 Target Size

</details>

**Comment by alastc** [comment]
Hi Dan,

WCAG 3.0 (code-named Silver) is likely to take quite a different approach, with multiple ways of organising the guidelines:
https://w3c.github.io/silver/requirements/#multiple-ways-to-display

In the 2.x line it would be very difficult to re-categorise them, that would cause more issues than it solves, unfortunately.

**Comment by Dan503** [comment]
Oh cool. Being able to organise them into different categories based on audience sounds great.

It's ok, i wasn't suggesting the 2.x criteria get rearranged. I just wanted to ensure that when WCAG 3.0 comes out eventually it doesn't have the same issue.

If you want you can close this or you might want to keep it open as a reference for 3.0.

**Comment by alastc** [comment]
Thanks Dan. There's a separate repo for WCAG 3, and it doesn't seem likely they will drop that requirement so it's probably best to just close it.
